core/common/src/rpc/web/WebAppRpcRequest.ts

```diff
@@ -284,9 +286,26 @@ export class WebAppRpcRequest extends RpcRequest {
 
   private static configureResponse(protocol: WebAppRpcProtocol, request: SerializedRpcRequest, fulfillment: RpcRequestFulfillment, res: HttpServerResponse) {
     const success = protocol.getStatus(fulfillment.status) === RpcRequestStatus.Resolved;
+    // TODO: Use stale-while-revalidate in cache headers. This needs to be tested, and does not currently have support in the router/caching-service.
+    // This will allow browsers to use stale cached responses while also revalidating with the router, allowing us to start up a backend if necessary.
 
+    // RPC Caching Service uses the s-maxage header to determine the TTL for the redis cache.
+    const oneHourInSeconds = 3600;
     if (success && request.caching === RpcResponseCacheControl.Immutable) {
-      res.set("Cache-Control", "private, max-age=31536000, immutable");
+      // If response size is > 50 MB, do not cache it.
+      if (fulfillment.result.objects.length > (50 * 10**7)) {
+        res.set("Cache-Control", "no-store");
+      } else if (request.operation.operationName === "generateTileContent") {
+        res.set("Cache-Control", "no-store");
+      } else if (request.operation.operationName === "getConnectionProps") {
+        // GetConnectionprops can't be cached on the browser longer than the lifespan of the backend. The lifespan of backend may shrink too. Keep it at 1 second to be safe.
```

Comment:
I think this would be better handled by setting it on the RPC methods directly instead of buried in this file. It would allow for better visibility on how each is cached.

Reply:
Yeah I agree, I made changes to support setting it directly on the RPC methods by just adding to the existing allowResponseCaching decorator. This will let frontends control their cache-control headers though, not sure if thats a problem or not. I was also unsure if "no-store" should be the default or not when no cache control header is passed to the decorator, but it seemed like the safest option.

What do you think?

Reply:
>  I was also unsure if "no-store" should be the default or not when no cache control header is passed to the decorator, but it seemed like the safest option.

I agree with this as the default as that's what it is technically today.

> This will let frontends control their cache-control headers though, not sure if thats a problem or not
Isn't that technically how it will always work?  The frontend is always in charge of what type of request it sends and the headers it could use.  With the way you had implemented before that would have just required someone to create their own sub-class of WebAppRpcRequest and override, I think. Is that correct or am i mis-interpreting it?

@wgoehrig for his input here...

Reply:
The header that is currently being set on the frontend, is actually the header that the backend is going to use when sending back the response (as long as it was successful and caching is enabled) through the caching service and eventually back to the frontend.

Also, for any currently exisiting frontends which have converted POSTs to GETs (I'm guessing there are very few). They would've been relying on the old cache header which was max-age=(3600 * 8700), 8700 hours of caching. Technically with this change, the new default would be the no-store for them.

Reply:
Okay, I think that's fine. As you said there were very few in rpcs that used GET in the first place and probably didn't know that the max-age was even being set on their behalf.

Also reporting a question I had above but it wasn't properly formatted...

> This will let frontends control their cache-control headers though, not sure if thats a problem or not

Isn't that technically how it will always work?  The frontend is always in charge of what type of request it sends and the headers it could use.  With the way you had implemented before that would have just required someone to create their own sub-class of WebAppRpcRequest and override, I think. Is that correct or am i mis-interpreting it?

Reply:
I thought that with how it was originally implemented, it would've required a change in the backend in order to change the cache headers, because configureResponse would've only been called by the backend code.

Reply:
@calebmshafer I ended up reverting the changes...Bill told me that you and him spoke about this and we should stick with keeping it in ConfigureResponse for the time being and eventually move to support it on the impl side, that way the backend stays in control.

Reply:
Sounds good, would you mind filing a backlog item for us to re-visit this and add the ability to add the ability to have decorators on the backend?

Reply:
done